gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java

```diff
@@ -589,6 +589,24 @@ public FSDataInputStream open(Path hadoopPath, int bufferSize) throws IOExceptio
         });
   }
 
+  public FSDataInputStream open(FileStatus status) throws IOException {
```

Comment:
Why FileStatus and not directly use GoogleHadoopFileStatus given it's a custom api?

Reply:
This can be called from outside connector and it may not have GoogleHadoopFileStatus dependency.

Reply:
I would disagree. This is a custom API we are creating if someone is calling it they have to have gcs-connector library to use it and hence they would have `GoogleHadoopFileStatus`.

Reply:
Discussed offline, it's fine to use `FileStatus` given the way we are planning to use it. `GoogleHadoopFileStatus` will make it more complex to use.